java-checks/src/main/java/org/sonar/java/checks/MathClampMethodsCheck.java

```diff
@@ -0,0 +1,105 @@
+    return version.isJava21Compatible();
+  }
+
+  @Override
+  public List<Tree.Kind> nodesToVisit() {
+    return List.of(Tree.Kind.CONDITIONAL_EXPRESSION, Tree.Kind.METHOD_INVOCATION, Tree.Kind.IF_STATEMENT);
+  }
+
+  @Override
+  public void visitNode(Tree tree) {
+    if (tree.is(Tree.Kind.CONDITIONAL_EXPRESSION)) {
+      checkConditionalExpression((ConditionalExpressionTree) tree);
+    }
+    if (tree.is(Tree.Kind.METHOD_INVOCATION)) {
+      checkMethodInvocation((MethodInvocationTree) tree);
+    }
+    if (tree.is(Tree.Kind.IF_STATEMENT)) {
+      checkIfStatement((IfStatementTree) tree);
+    }
+  }
+
+  private void checkConditionalExpression(ConditionalExpressionTree tree) {
+    if (isGreaterThanOrLessThan(tree.condition()) && (areConditionalOrMathMethodInvocation(tree.trueExpression(), tree.falseExpression()))) {
+      reportIssue(tree, "Use \"Math.clamp\" instead of a conditional expression.");
+    }
+  }
+
+  private void checkMethodInvocation(MethodInvocationTree tree) {
+    var mit = getMethodInvocationTree(tree);
+    if (mit != null && (areConditionalOrMathMethodInvocation(mit.arguments().get(0), mit.arguments().get(1)))) {
+      reportIssue(mit, "Use \"Math.clamp\" instead of \"Math.min\" or \"Math.max\".");
+    }
+  }
+
+  private static boolean areConditionalOrMathMethodInvocation(ExpressionTree tree1, ExpressionTree tree2) {
+    return (tree1.is(Tree.Kind.CONDITIONAL_EXPRESSION) && isGreaterThanOrLessThan(((ConditionalExpressionTree) tree1).condition()))
+      || (tree2.is(Tree.Kind.CONDITIONAL_EXPRESSION) && isGreaterThanOrLessThan(((ConditionalExpressionTree) tree2).condition()))
+      || (tree1.is(Tree.Kind.METHOD_INVOCATION) && getMethodInvocationTree((MethodInvocationTree) tree1) != null)
+      || (tree2.is(Tree.Kind.METHOD_INVOCATION) && getMethodInvocationTree((MethodInvocationTree) tree2) != null);
+  }
+
+  private static MethodInvocationTree getMethodInvocationTree(MethodInvocationTree mit) {
+    if (mit.methodSelect().is(Tree.Kind.MEMBER_SELECT)) {
+      var memberSelectExpressionTree = (MemberSelectExpressionTree) mit.methodSelect();
+      if (memberSelectExpressionTree.expression().symbolType().is("java.lang.Math")
+        && ("min".equals(memberSelectExpressionTree.identifier().name()) || "max".equals(memberSelectExpressionTree.identifier().name()))) {
+        return mit;
+      }
+    }
+    return null;
+  }
+
+  private void checkIfStatement(IfStatementTree tree) {
+    if (isGreaterThanOrLessThan(tree.condition())
+      && (tree.elseStatement() != null && tree.elseStatement().is(Tree.Kind.IF_STATEMENT))) {
+      var elseIfStatement = (IfStatementTree) tree.elseStatement();
+      if (isGreaterThanOrLessThan(elseIfStatement.condition())) {
+        reportIssue(tree, "Use \"Math.clamp\" instead of an if-else statement.");
+      }
+    }
+  }
+
+  private static boolean isGreaterThanOrLessThan(Tree tree) {
+    return tree.is(Tree.Kind.GREATER_THAN) || tree.is(Tree.Kind.LESS_THAN);
```

Comment:
We should consider GREATER_THAN_OR_EQUAL_TO like GREATER_THAN.

And be careful:
`Math.max(nextValue(), Math.min(nextValue(), value))`==
`Math.clamp(value, nextValue(), nextValue())`
But
`Math.min(nextValue(), Math.max( nextValue(), value))` !=
`Math.clamp(value, nextValue(), nextValue())`